wherehows-web/app/routes/datasets/dataset/compliance.ts

```diff
@@ -1,32 +1,30 @@
-import { set } from '@ember/object';
-import { descendantDatasetRouteClassFactory, Tabs } from 'wherehows-web/constants/datasets/shared';
+import {
+  descendantDatasetRouteClassFactory,
+  setPropertyOnAncestorController,
+  Tabs
+} from 'wherehows-web/constants/datasets/shared';
 import DatasetComplianceController from 'wherehows-web/controllers/datasets/dataset/compliance';
 import { refreshModelForQueryParams } from 'wherehows-web/utils/helpers/routes';
-import { IDatasetView } from 'wherehows-web/typings/api/datasets/dataset';
+import { isTagFilter, TagFilter } from 'wherehows-web/constants';
 import DatasetController from 'wherehows-web/controllers/datasets/dataset';
 
 export default class Compliance extends descendantDatasetRouteClassFactory({ selectedTab: Tabs.Compliance }) {
   queryParams = refreshModelForQueryParams(['fieldFilter']);
 
-  setupController(controller: DatasetComplianceController, model: IDatasetView): void {
-    const { setPropOnAncestorController } = this;
+  async model(): Promise<DatasetController['model'] | object> {
+    return { ...this.controllerFor('datasets.dataset').model };
+  }
+
+  setupController(controller: DatasetComplianceController, model: object): void {
+    const { fieldFilter } = controller;
 
     super.setupController(controller, model);
 
-    // set the compliance tagFilter on the ancestor controller
-    setPropOnAncestorController('complianceTagFilter', controller.fieldFilter);
+    // set the complianceTagFilter property on the ancestor controller
+    setPropertyOnAncestorController(
```

Comment:
Again, setting properties in a controller vs model. I think we should discourage settings properties in a controller since controllers are singletons, where as a model is not. Also, it is kind of coupling parent with child. I think child depends on parent is ok, but not ok in the opposite direction? thoughts?

Reply:
Debatable. The model for the dataset route conforms to the interface `IDatasetView`, i'm definitely against augmenting the model to add client side information that is not related to the route's model interface, separation of concerns.

That, to me, is the purpose of the controller, for things like shared configuration data and queryParams. `complianceTagFilter` is a queryParam, and above `isInternal` etc are configuration values that make sense to live on the controller. 
I think the model should only hold business logic concerns, and controllers can host application/infrastructure logic.

To your second point, I don't think you can even do parent to child Ember will throw and error because the child is not even instantiated, except you manually do that ofcourse